plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/config/CPSConfiguration.java

```diff
@@ -24,32 +24,58 @@
 
 package org.jenkinsci.plugins.workflow.cps.config;
 
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
 import edu.umd.cs.findbugs.annotations.NonNull;
 import hudson.Extension;
 import hudson.ExtensionList;
 import jenkins.model.GlobalConfiguration;
 import jenkins.model.GlobalConfigurationCategory;
+import org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval;
 import org.jenkinsci.Symbol;
 
+/**
+* @deprecated
+ * This class hs been deprecated and we don't recommend to use it.
+ * In order to force using the system sandbox for pipelines, please use the flag
+ * org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval.get().isForceSandbox
+ * or
+ * org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval.get().forceSandboxForCurrentUser
+ **/
 @Symbol("cps")
 @Extension
+@Deprecated
 public class CPSConfiguration extends GlobalConfiguration {
 
     /**
      * Whether to show the sandbox checkbox in jobs to users without Jenkins.ADMINISTER
      */
-    private boolean hideSandbox;
+    private transient boolean hideSandbox;
 
     public CPSConfiguration() {
+
         load();
+        if (hideSandbox) {
+            ScriptApproval.get().setForceSandbox(hideSandbox);
+        }
+
+        //Data migration from this configuration to ScriptApproval should be done only once,
+        //so removing the config file after the previous migration
+        try {
+            this.getConfigFile().delete();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
     }
 
     public boolean isHideSandbox() {
-        return hideSandbox;
+        return ScriptApproval.get().isForceSandbox();
     }
 
     public void setHideSandbox(boolean hideSandbox) {
         this.hideSandbox = hideSandbox;
+        ScriptApproval.get().setForceSandbox(hideSandbox);
         save();
     }
```

Comment:
I think you can remove this 2 methods and update their usages to call the `ScriptApproval` ones directly.

Reply:
Basically, the class stays just for the migration, nothing else.

Reply:
Ok, will remove them.